unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Loader/Loader.cs

```diff
@@ -0,0 +1,214 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Cysharp.Threading.Tasks;
+using DCL;
+using DCL.Helpers;
+using UnityEngine;
+
+namespace AvatarSystem
+{
+    public class Loader : ILoader
+    {
+        public GameObject bodyshapeContainer => bodyshapeLoader?.rendereable?.container;
+        public SkinnedMeshRenderer combinedRenderer { get; private set; }
+        public Renderer[] facialFeaturesRenderers { get; private set; }
+        public ILoader.Status status { get; private set; } = ILoader.Status.Idle;
+
+        private readonly IWearableLoaderFactory wearableLoaderFactory;
+        private readonly GameObject container;
+
+        internal IBodyshapeLoader bodyshapeLoader;
+        internal readonly Dictionary<string, IWearableLoader> loaders = new Dictionary<string, IWearableLoader>();
+        private readonly IAvatarMeshCombinerHelper avatarMeshCombiner;
+
+        public Loader(IWearableLoaderFactory wearableLoaderFactory, GameObject container, IAvatarMeshCombinerHelper avatarMeshCombiner)
+        {
+            this.wearableLoaderFactory = wearableLoaderFactory;
+            this.container = container;
+
+            this.avatarMeshCombiner = avatarMeshCombiner;
+            avatarMeshCombiner.prepareMeshForGpuSkinning = true;
+            avatarMeshCombiner.uploadMeshToGpu = true;
+        }
+
+        public async UniTask Load(WearableItem bodyshape, WearableItem eyes, WearableItem eyebrows, WearableItem mouth, List<WearableItem> wearables, AvatarSettings settings, CancellationToken ct = default)
+        {
+            ct.ThrowIfCancellationRequested();
+
+            List<IWearableLoader> toCleanUp = new List<IWearableLoader>();
+
+            void DisposeCleanUpLoaders()
+            {
+                for (int i = 0; i < toCleanUp.Count; i++)
+                {
+                    if (toCleanUp[i] == null)
+                        continue;
+                    toCleanUp[i].Dispose();
+                }
+            }
+
+            try
+            {
+                status = ILoader.Status.Loading;
+
+                if (bodyshapeLoader == null || bodyshapeLoader.wearable.id != bodyshape.id || bodyshapeLoader.eyes.id != eyes.id || bodyshapeLoader.eyebrows.id != eyebrows.id || bodyshapeLoader.mouth.id != mouth.id)
```

Comment:
Can we add a comment here explaining this big if? or move it to a function to improve readability